security/pkg/pki/ca/ca.go

```diff
@@ -184,14 +187,21 @@ func (ca *IstioCA) Sign(csrPEM []byte) ([]byte, error) {
 	return chain, nil
 }
 
-func (ca *IstioCA) generateCertificateTemplate(request *x509.CertificateRequest) *x509.Certificate {
+func (ca *IstioCA) generateCertificateTemplate(request *x509.CertificateRequest, requestedTTL time.Duration) *x509.Certificate {
 	exts := append(request.Extensions, request.ExtraExtensions...)
+
+	certTTL := requestedTTL
+	// If the requested TTL is greater than maxCertTTL, apply maxCertTTL as the TTL.
+	if requestedTTL.Seconds() > ca.maxCertTTL.Seconds() {
```

Comment:
I'm wondering whether you should reject the request instead of resetting it to max?

Reply:
That makes sense. I can make it return an error.